### DISCUSSION TITLE
//SOURCES **.*.java behaves differently for local and remote execution

### DISCUSSION THREAD

**Discussion by Sandared** [jbangdev/jbang#1621] [Q&A] [answered]
If I have two classes e.g.,
```java
//SOURCES **.*.java
//DEPS some:deps:1.0.0
package io.test.folder;
public class A {
  private B b;
  ...
}

---

package io.test.folder.subfolder;
public class B {}
```
and I invoke `jbang A.java` locally then everything works fine. 
However, if I do the same in a remote context, e.g., `jbang myACode@test-io` and this is an alias for the github repo where the code lies then jbang is not able to compile this, as it misses B.java.

Is this an intended behavior?
If so, is there a workaround for this?

Kind regards,
Thomas

**Comment by quintesse** [comment, marked as answer]
Hi @Sandared , well it's not so much intended behaviour but simply the fact that's the only behaviour possible when dealing with remote sources because most of the time there's no way to turn `**/*.java` into an actual list. (Now, we might add support for sites like GitHub where we _can_ list files at some point in the future, but there are no real plans for that yet).

So is there a work-around? Yes: don't use patterns, simply explicitly list all the files one by one. This will then work both locally and remotely.

**Reply by Sandared** [reply]
Hey @quintesse  
Thanks for the quick answer :). Makes sense, but I had hoped I get around writing down all classes... lazy me :P

**Reply by maxandersen** [reply]
I have considered making a your file.java.idx which could have the paths matched on a build that one could generate using `jbang export index` that jbang would use if present.

**Reply by Sandared** [reply]
That would definitely be an improvement for cases like mine (having like 30 .java files to write down) :)

**Reply by Sandared** [reply]
BTW: has nothing to do with this issue, but just wanted to say it: JBang is awesome... you created something really really cool 👍😁
